file.h

```diff
@@ -1039,105 +1046,10 @@ std::size_t basic_file_base<File>::write(cbyte_view data) const
 template <typename File>
 std::vector<std::byte> basic_file_base<File>::read(std::size_t size) const
 {
-    // In the absence of size and file size, this will be the initial size
-    // of the vector.
-    constexpr std::size_t initial_vector_size = 0x1000;
-
-    // The file data.
-    std::vector<std::byte> data;
-
-    // If size is zero, we need to read the entire file,
-    // try to reserve space according to the file size.
-    if (!size) {
-        // Get the file size.
-        auto file_size = this->size();
-
-        // If file size cannot be determined, resize to the initial size.
-        if (!file_size) {
-            // Resize to initial size.
-            data.resize(initial_vector_size);
-        } else {
-            // Get the file offset.
-            auto current_offset = tell();
-
-            // If the offset is beyond the size, return immediately.
-            if (current_offset >= file_size) {
-                return data;
-            }
-
-            // Compute the amount of bytes to read.
-            auto data_size = file_size - current_offset;
-
-            // Check whether the data size is larger than the
-            // implementation capacity.
-            if (data_size > (std::numeric_limits<std::size_t>::max)()) {
-                throw std::range_error(
-                    "Amount of file data is too large for this platform.");
-            }
-
-            // Resize to the data size.
-            data.resize(static_cast<std::size_t>(data_size));
-        }
-    } else {
-        // Reserve the requested size.
-        data.resize(size);
+    if (size == 0) {
```

Comment:
I am taking this idea and have two read overloads, one for reading the entire file and one for reading a specific size.

Reply:
Good. But sometimes I think it is better to be explicit. 
I propose read_all().
This is similar to the train of thought that encourages us to write explicit static factories instead of relying on overloaded constructors. Sometimes you can get away with it, but often the variety of argument types is not enough to describe what the method does.